File: benchmark/corpus/python/path_vuln_1.py
"""VULN: CWE-22 — Path traversal via open() with unsanitised user filename."""
import os
import flask

app = flask.Flask(__name__)
BASE_DIR = "/var/www/files"


@app.route("/download")
def download():
    filename = flask.request.args.get("file", "")
    filepath = os.path.join(BASE_DIR, filename)
    with open(filepath, "rb") as f:
        return f.read()

File: benchmark/corpus/javascript/path_vuln_1.js
// VULN: CWE-22 — Path traversal via path.join() with user-controlled filename
const express = require('express');
const fs = require('fs');
const path = require('path');
const app = express();

const BASE_DIR = '/var/www/files';

app.get('/download', (req, res) => {
    const filename = req.query.file;
    const filepath = path.join(BASE_DIR, filename);
    fs.readFile(filepath, (err, data) => {
        if (err) return res.status(404).send('Not found');
        res.send(data);
    });
});

app.listen(3000);
